packages/app/src/utils/parseSignature.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * Parse a function signature string from OpenChain/Sourcify into structured format
+ * Example: "transfer(address,uint256)" -> { name: "transfer", inputs: [...] }
+ */
+
+export interface ParsedSignature {
+  name: string;
+  inputs: Array<{
+    type: string;
+    name: string;
+  }>;
+}
+
+/**
+ * Parse a function signature string into a structured format
+ * @param signature - Function signature like "transfer(address,uint256)"
+ * @returns Parsed signature object or null if invalid
+ */
+export function parseSignature(signature: string): ParsedSignature | null {
```

Comment:
Why not using FunctionFragment from ethers? `FunctionFragment.from("transfer(address,uint256)");`